xcodebuild.go

```diff
@@ -234,11 +241,19 @@ func createXcodebuildTestArgs(params models.XcodeBuildTestParamsModel, xcodeMajo
 		xcodebuildArgs = append(xcodebuildArgs, "GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES", "GCC_GENERATE_TEST_COVERAGE_FILES=YES")
 	}
 
-	if xcodeMajorVersion >= 13 && params.RetryTestsOnFailure {
+	switch params.TestRepetitionMode {
+	case untilFailure:
+		xcodebuildArgs = append(xcodebuildArgs, "-run-tests-until-failure")
+	case retryOnFailure:
 		xcodebuildArgs = append(xcodebuildArgs, "-retry-tests-on-failure")
+	case upUntilMaximumRepetitions, none:
+		break
+	default:
+		panic("Invalid test repetition mode: " + params.TestRepetitionMode)
```

Comment:
Would reccomend not using panic here as ussually we try to not exit, only from main(). By introducing an enum (similar to https://github.com/bitrise-steplib/steps-xcode-test/blob/master/inputs.go), that does not have an `invalid` option (and was previously validated), one can be sure that there is no possibility for an invalid case here.

Reply:
In that case, we can just remove the `default` branch here, because `TestRepetitionMode` was validated before by `stepconf.Parse()` (because `TestRepetitionMode` is annotated with `opt[...]`).